app/src/main/java/nodomain/freeyourgadget/gadgetbridge/service/devices/hplus/HPlusSupport.java

```diff
@@ -0,0 +1,881 @@
+    public void onNotification(NotificationSpec notificationSpec) {
+        LOG.debug("Got Notification");
+        showText(notificationSpec.body);
+    }
+
+
+    @Override
+    public void onSetTime() {
+        TransactionBuilder builder = new TransactionBuilder("vibration");
+        setCurrentDate(builder);
+        setCurrentTime(builder);
+
+    }
+
+    @Override
+    public void onSetAlarms(ArrayList<? extends Alarm> alarms) {
+
+    }
+
+    @Override
+    public void onSetCallState(CallSpec callSpec) {
+        switch (callSpec.command) {
+            case CallSpec.CALL_INCOMING: {
+                showText(callSpec.name, callSpec.number);
+                break;
+            }
+        }
+
+    }
+
+    @Override
+    public void onSetCannedMessages(CannedMessagesSpec cannedMessagesSpec) {
+        LOG.debug("Canned Messages: " + cannedMessagesSpec);
+    }
+
+    @Override
+    public void onSetMusicState(MusicStateSpec stateSpec) {
+
+    }
+
+    @Override
+    public void onSetMusicInfo(MusicSpec musicSpec) {
+
+    }
+
+    @Override
+    public void onEnableRealtimeSteps(boolean enable) {
+
+    }
+
+    @Override
+    public void onInstallApp(Uri uri) {
+
+    }
+
+    @Override
+    public void onAppInfoReq() {
+
+    }
+
+    @Override
+    public void onAppStart(UUID uuid, boolean start) {
+
+    }
+
+    @Override
+    public void onAppDelete(UUID uuid) {
+
+    }
+
+    @Override
+    public void onAppConfiguration(UUID appUuid, String config) {
+
+    }
+
+    @Override
+    public void onAppReorder(UUID[] uuids) {
+
+    }
+
+    @Override
+    public void onFetchActivityData() {
+
+    }
+
+    @Override
+    public void onReboot() {
+
+    }
+
+    @Override
+    public void onHeartRateTest() {
+        LOG.debug("On HeartRateTest");
+
+        getQueue().clear();
+
+        TransactionBuilder builder = new TransactionBuilder("HeartRateTest");
+        byte state = 0;
+
+        builder.write(ctrlCharacteristic, new byte[]{HPlusConstants.COMMAND_SET_PREF_ALLDAYHR, 0x10}); //Set Real Time... ?
+        builder.queue(getQueue());
+
+    }
+
+    @Override
+    public void onEnableRealtimeHeartRateMeasurement(boolean enable) {
+        LOG.debug("Set Real Time HR Measurement: " + enable);
+
+        getQueue().clear();
+
+        TransactionBuilder builder = new TransactionBuilder("realTimeHeartMeasurement");
+        byte state = 0;
+
+        if (enable)
+            state = HPlusConstants.HEARTRATE_ALLDAY_ON;
+        else
+            state = HPlusConstants.HEARTRATE_ALLDAY_OFF;
+
+        builder.write(ctrlCharacteristic, new byte[]{HPlusConstants.COMMAND_SET_PREF_ALLDAYHR, state});
+        builder.queue(getQueue());
+    }
+
+    @Override
+    public void onFindDevice(boolean start) {
+        LOG.debug("Find Me");
+
+        getQueue().clear();
+        ctrlCharacteristic = getCharacteristic(HPlusConstants.UUID_CHARACTERISTIC_CONTROL);
+
+        TransactionBuilder builder = new TransactionBuilder("findMe");
```

Comment:
Typically we use performInitialized() to acquire a TransactionBuilder(). This will attempt to connect and initialize the device prior to executing the action (here: "find device").